pkg/wait/conditions.go

```diff
@@ -149,6 +150,35 @@ func IsAppStatus(ctx context.Context, kubeClient *client.Client, appName string,
 	}
 }
 
+// IsAllAppStatus returns a WaitCondition that checks if all the apps provided currently have the provided expected status
+func IsAllAppStatus(ctx context.Context, kubeClient *client.Client, appNamespacedNames []types.NamespacedName, expectedStatus string) WaitCondition {
+	return func() (bool, error) {
+		var err error
+		isSuccess := true
+
+		for _, namespacedName := range appNamespacedNames {
+			app := &applicationv1alpha1.App{
+				ObjectMeta: v1.ObjectMeta{
+					Name:      namespacedName.Name,
+					Namespace: namespacedName.Namespace,
+				},
+			}
+			if err = kubeClient.Client.Get(ctx, cr.ObjectKeyFromObject(app), app); err != nil {
+				isSuccess = false
+				continue
+			}
+
+			actualStatus := app.Status.Release.Status
+			logger.Log("Checking if App status for %s is equal to '%s': %s", namespacedName.Name, expectedStatus, actualStatus)
+			if expectedStatus != actualStatus {
+				isSuccess = false
```

Comment:
A small optimisation might be to directly return false instead of using `isSuccess` to save time and limit api calls.

Same above, does it make sense to continue if you already know it's a failure?

Reply:
Yes because then the logs give us full insight into what apps are ready and what are not.